src/nrnpython/grids.cpp

```diff
@@ -415,21 +415,21 @@ extern "C" int ICS_insert(int grid_list_index,
     return new_Grid->insert(grid_list_index);
 }
 
-int ICS_insert_inhom(int grid_list_index,
-                     PyHocObject* my_states,
-                     long num_nodes,
-                     long* neighbors,
-                     long* x_line_defs,
-                     long x_lines_length,
-                     long* y_line_defs,
-                     long y_lines_length,
-                     long* z_line_defs,
-                     long z_lines_length,
-                     double* dcs,
-                     double dx,
-                     bool is_diffusable,
-                     double atolscale,
-                     double* ics_alphas) {
+extern "C" int ICS_insert_inhom(int grid_list_index,
```

Comment:
For clarity it might be a good idea to set these extra attributes in the function declaration only.

Reply:
I tried but this is neuron, there is no clear concepts of header files. Declarations/definitions. And it was failing sometimes.

We can in a new PR try to clean this code and split things in "module" (good luck).